lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -95,15 +98,26 @@ export class MsalInterceptor implements HttpInterceptor {
      * @param scopes Array of scopes for the request
      * @returns Result from the interactive request
      */
-    private acquireTokenInteractively(authRequest: MsalInterceptorAuthRequest, scopes: string[]): Observable<AuthenticationResult> {
-        if (this.msalInterceptorConfig.interactionType === InteractionType.Popup) {
-            this.authService.getLogger().verbose("Interceptor - error acquiring token silently, acquiring by popup");
-            return this.authService.acquireTokenPopup({ ...authRequest, scopes });
-        }
-        this.authService.getLogger().verbose("Interceptor - error acquiring token silently, acquiring by redirect");
-        const redirectStartPage = window.location.href;
-        this.authService.acquireTokenRedirect({...authRequest, scopes, redirectStartPage });
-        return EMPTY;
+    private acquireTokenInteractively(authRequest: MsalInterceptorAuthRequest, scopes: string[]): Observable<AuthenticationResult> { 
+        var subject = new Subject<AuthenticationResult>();
+        this.msalBroadcastService.inProgress$
```

Comment:
I think potentially a better approach is to have the interceptor class start listening for inProgress events in the constuctor, and main an internal field (i.e. this.currentInteractionStatus), instead of listening only when acquireTokenInteractively is called.

Reply:
Agreed. I also think it would be good to have it call acquireTokenSilent again when inProgress turns back to `None`. This would prevent a scenario where many calls are queued up all invoking their own interaction back to back to back.